lib/cli/run-helpers.js

```diff
@@ -79,62 +79,31 @@ exports.list = str =>
  *
  * Returns array of `mochaHooks` exports, if any.
  * @param {string[]} requires - Modules to require
- * @returns {Promise<MochaRootHookObject|MochaRootHookFunction>} Any root hooks
+ * @returns {Promise<object>} Plugin implementations
  * @private
  */
 exports.handleRequires = async (requires = []) => {
-  const acc = [];
-  for (const mod of requires) {
+  const pluginLoader = PluginLoader.create();
+  for await (const mod of requires) {
     let modpath = mod;
     // this is relative to cwd
     if (fs.existsSync(mod) || fs.existsSync(`${mod}.js`)) {
       modpath = path.resolve(mod);
       debug('resolved required file %s to %s', mod, modpath);
     }
     const requiredModule = await requireOrImport(modpath);
-    if (
-      requiredModule &&
-      typeof requiredModule === 'object' &&
-      requiredModule.mochaHooks
-    ) {
-      const mochaHooksType = type(requiredModule.mochaHooks);
-      if (/function$/.test(mochaHooksType) || mochaHooksType === 'object') {
-        debug('found root hooks in required file %s', mod);
-        acc.push(requiredModule.mochaHooks);
-      } else {
-        throw createUnsupportedError(
-          'mochaHooks must be an object or a function returning (or fulfilling with) an object'
-        );
+    if (requiredModule && typeof requiredModule === 'object') {
```

Comment:
the logic here is "look in the module we've loaded via `--require` for any known named exports and figure out what to do with them".  since these could be multiple and spread out over several files, on L102 below, we call `pluginLoader.finalize()` which allows aggregation of these into a single object, suitable for passing into the `Mocha` constructor.  

for example, we might have `mochaHooks` in `a.js` and `mochaHooks` in `b.js`; the `pluginLoader.load()` call will _collect_ these, and we mash them all together into an object (containing props for each hook type, and arrays of hook callback functions) in `pluginLoader.finalize()`.